IPython/core/magics/pylab.py

```diff
@@ -84,9 +86,15 @@ def matplotlib(self, line=''):
 
             In [3]: %matplotlib qt
         """
+
         args = magic_arguments.parse_argstring(self.matplotlib, line)
-        gui, backend = self.shell.enable_matplotlib(args.gui)
-        self._show_matplotlib_backend(args.gui, backend)
+
+        if args.list is not None:
```

Comment:
I think this should just be `if args.list`, then you shouldn't need the default above (with `store_true` the default should automatically be False).